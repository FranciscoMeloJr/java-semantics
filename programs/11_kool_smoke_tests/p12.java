class main {
  int f(int x) {
    if (x <= 1) return 1; else return(x * (f(x - 1)));
  }
  void main(String[] args) {
    System.out.print(f(f(4)));
  }
}

