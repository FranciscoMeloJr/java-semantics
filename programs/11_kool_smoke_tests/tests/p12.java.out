-775946240
Done!
